### PR TITLE
[Cloud] Fix cloud set name case sensitive

### LINKED ISSUE #4663: `az cloud set -n` should not be case sensitive
### Description
While *bash* is case sensitive, and the concept makes sense to those users, I don't see a business or customer value in make this a policy.  As a result, having cloud names be case sensitive only serves to add another error condition.

```
az cloud set -n azurecloud
The cloud 'azurecloud' is not registered.
[1] ~ $ az cloud set -n AzureCloud
Switched active cloud to 'AzureCloud'.
```

### PR DESCRIPTION
**Pull request by limingu**
Fixes #4663
---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

File: src/azure-cli-core/azure/cli/core/cloud.py [hunk 1]
```diff
@@ -256,6 +256,8 @@ def get_custom_clouds(cli_ctx):
     known_cloud_names = [c.name for c in KNOWN_CLOUDS]
     return [c for c in get_clouds(cli_ctx) if c.name not in known_cloud_names]
 
+def get_cloud_name(cli_ctx, cloud_name):
+    return next((x.name for x in get_clouds(cli_ctx) if x.name.lower() == cloud_name.lower()), cloud_name)
```
Comment (tjprescott) [inline src/azure-cli-core/azure/cli/core/cloud.py]: Let's make this private so as not to expand the public interface of core. `_get_cloud_name`

File: src/command_modules/azure-cli-cloud/azure/cli/command_modules/cloud/tests/latest/test_cloud.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+    def test_cloud_set_azureusgovernment(self):
+        self.cmd('az cloud set -n azureusgovernment')
+
+    def test_cloud_set_azuregermancloud(self):
+        self.cmd('az cloud set -n azuregermancloud')
+
+    def test_cloud_set_unregistered_cloud_name(self):
+        with pytest.raises(Exception, match =  "The cloud 'azCloud' is not registered.") :
+            self.cmd('az cloud set -n azCloud').get_output_in_json()
```
Comment (tjprescott) [inline src/command_modules/azure-cli-cloud/azure/cli/command_modules/cloud/tests/latest/test_cloud.py]: I'd recommend following the syntax here:
https://github.com/Azure/azure-cli/blob/dev/doc/authoring_tests.md#sample-9-assert-specific-error-occurs

This avoids needing a new dependency on pytest.

File: src/azure-cli-core/azure/cli/core/cloud.py (hunk 1 above)
Comment (limingu) [reply src/azure-cli-core/azure/cli/core/cloud.py]: Updated

File: src/command_modules/azure-cli-cloud/azure/cli/command_modules/cloud/tests/latest/test_cloud.py (hunk 2 above)
Comment (limingu) [reply src/command_modules/azure-cli-cloud/azure/cli/command_modules/cloud/tests/latest/test_cloud.py]: Updated

**Pull request merged**